tools/linter/clang_tidy/__main__.py

```diff
@@ -185,15 +178,16 @@ def main() -> None:
 
     if not exists:
         msg = (
-            "Could not find '.clang-tidy-bin/clang-tidy'\n"
-            "You can install it by running:\n"
-            "   python3 tools/linter/install/clang_tidy.py"
+            f"Could not find '{options.clang_tidy_exe}'\n"
+            + "We provide a custom build of clang-tidy that has additional checks.\n"
+            + "You can install it by running:\n"
+            + "$ python3 tools/linter/install/clang_tidy.py"
         )
         raise RuntimeError(msg)
 
     return_code = run(options)
     if return_code != 0:
-        raise RuntimeError("Warnings found in clang-tidy output!")
+        raise RuntimeError("Your code is not clang-tidy clean!")
```

Comment:
The Python traceback isn't super useful for a failure in this case imo, I think it'd be better to just exit with an error like mypy/flake8

```suggestion
        sys.exit(returncode)
```